modules/gapi/include/opencv2/gapi/core.hpp

```diff
@@ -1342,10 +1352,26 @@ enlarge an image, it will generally look best with cv::INTER_CUBIC (slow) or cv:
 \f[\texttt{(double)dsize.height/src.rows}\f]
 @param interpolation interpolation method, see cv::InterpolationFlags
 
-@sa  warpAffine, warpPerspective, remap
+@sa  warpAffine, warpPerspective, remap, resizeP
  */
 GAPI_EXPORTS GMat resize(const GMat& src, const Size& dsize, double fx = 0, double fy = 0, int interpolation = INTER_LINEAR);
 
+/** @brief Resizes a planar image.
+
+The function resizes the image src down to or up to the specified size.
+
+Output image size will have the size dsize, the depth of output is the same as of src.
+
+@note Function textual ID is "org.opencv.core.transform.resizeP"
+
+@param src input image, must be of @ref CV_8UC3 type;
```

Comment:
> planar image
> CV_8UC3

`CV_8UC3` means interleaved image. So `planar` is source of confusion here.
planar is src1, src2 and src3 with `CV_8UC1` type.

What is the main difference between `resize` and `resizeP`? Missing `fx`, `fy`?
Do we really need this entry point?

---

How do you want to downscale image by 2 with odd source size?
9x9 => 4.5 x 4.5 as 4x4 - is not 2x downscaling without extra information.

Reply:
The main difference is that `resizeP` has `GMatP` ("planar GMat") as input and as output so the engine can behave correctly.
This description definitely should be changed to reflect planar image memory layout (three planes laying in the memory contiguously), so the image height should be `plane_height`*`plane_number`, image type is `CV_8UC1` and so on. Maybe planar images deserve some special chapter?